### ISSUE TITLE
Allowed/not allowed content on NewPipe

### ISSUE THREAD

**Issue by theScrabi** [TeamNewPipe/documentation#3]
@TobiGr @TheAssassin I need your opinion about this. What would be the Lawful right thing to declare NOT accepted content?

https://teamnewpipe.github.io/documentation/00_Prepare_everything/#inclusion-criteria-for-services

**Comment by TheAssassin** [comment]
What *exactly* do you want to know? You basically created a list of content that you don't want in NewPipe, right?

I suppose you want contributors to agree to these conditions. Therefore, you need to set up a "contribution agreement" contract-ish document that needs to be accepted by contributors. I doubt it's legally valid, as the author of a plugin cannot be held liable in any way for the service they create a plugin for, and if you would want to enforce _such_ a rule, nobody sane would ever agree to such a contract. I mean, I wouldn't want to be sued for invalid content on any such platform for a software I contributed to, even _if_ that was legally possible.

What you _can_ do is enforce the exclusion of any service that violates your list, by reviewing new services. But any service that exists at the moment violates _at least_

> Copyright infringement/pirated media

so I would be _really_ careful about the points in this list.

> Any form of violence
> Violations of the human rights

is not at all very precise. Imagine a movie that shows torture. Is that covered by the freedom of art? At the moment, if any of the services contained such a movie, it would violate your list of conditions. Same goes for music titles that "glorify" violence, there's a lot of such music across all genres.

I bet there even has been pornography on YouTube, and I bet in every second, someone tries to upload such a video, considering the massive amount of material uploaded there, and I bet that YouTubes "auto filtering" stuff that is glorified by the people, sometimes press and especially the [European Commission](https://netzpolitik.org/2018/upload-filter-breites-buendnis-warnt-vor-gefaehrlichem-irrweg/), miss at least a little fraction of these videos.

I think that this is a highly complex topic that needs some long discussion, and you need to be really careful about what kind of rules you're trying to set up here. How you word these points is _really_ important, and highly difficult. I think even a person who is familiar with international law would have difficulties to take everything into account that is relevant, e.g., violation of the human rights, freedom of art, where they apply and where they don't, etc.

Also, you can't really enforce this list code wise. You can only control what is committed to this repository, but that's it. The code is still free/libre. Anyone can do almost anything with it, as long as they conform to the GPL. And the GPL doesn't prohibit any kind of "content filtering", which is intentional.

**Comment by theScrabi** [comment]
Yea well than lets take it a s a guideline which services will be included in our repositorry, and which not.

**Comment by TheAssassin** [comment]
@theScrabi still, your wording contradicts the inclusion of YouTube, SoundCloud, etc. If, in the future, you reject a specific service for one of the reasons I referred to above (and possibly even others), you'd apply double standards. In that case, you can't others expect to take you serious on this. I doubt you'd want that.

**Comment by theScrabi** [comment]
What was your suggestion to handle this?

**Comment by TheAssassin** [comment]
I just put together a list of issues with your current approach. There is no suggestion.

**Comment by theScrabi** [comment]
Would you like to think about one?

**Comment by TobiGr** [comment]
I think we should open a ticket in the Extractor's repository which lists all services which have been requested and would be accepted. When people want to implement a service, they should fill a request to get their service included __before__ they start implementing it.
When there are any legal/ethical/whatever concerns we could discuss them there.
A ticket like this can also help to prevent the case two different people implementing the same service simultaneously.

Btw, there are two small typos in `documentation`: https://github.com/TeamNewPipe/documentation/blob/master/docs/index.md

**Comment by TheAssassin** [comment]
> Would you like to think about one?

Not really. I personally think most of these "contribution agreements" make very much sense. I just wanted to point out why your wording is suboptimal, and might lead to difficult discussions with contributors, considering there's a fair amount of capriciousness in the service selection if those rules were finally published.

**Comment by theScrabi** [comment]
> hey should fill a request to get their service included before they start implementing it.

I would not do that. Imagine someone claims to work on a service but later abandoned it, other might think he is still working on it and don't give it a shot them serfs. I would discuss the inclusion when someone sends a PR or mentions him self in an issue that he works on something.

**Comment by TheAssassin** [comment]
> Imagine someone claims to work on a service but later abandoned it,

@theScrabi that's another point I need to criticize. The world of open source doesn't work like "We accept your contribution, thanks, but would you sign this <contract> and guarantee us a <SLA>? You need to set up a testament, so that in case you die someone else maintains your code."

**Comment by theScrabi** [comment]
Yea I was not going to do that. We don't need someone to sign a contract for his code to merge in our repository. We simply merge it or we don't.

Maybe my list from above can be seen as a guideline for what will be merged and what will not.

**Comment by tbm** [comment]
I just found this ticket after reading the docs and finding something confusing:

> Any kind of pornography or NSFW content that does not violate US law. However, porn services will not be added to the official NewPipe app

So pornography is allowed but porn services aren't?  This seems like a contradiction to me. If porn is okay, why not porn services?

**Comment by theScrabi** [comment]
No. It says that porn is ok if it's not violating NSFW. This means if someone is willing to write a service for NewPipe his or her contribution will be taken into consideration. However in the final App such Services will not be included. In that case we will have to fork NewPipe like we did with [NewPipe-legecy](https://github.com/TeamNewPipe/NewPipe-legacy) in order to keep pornographic contend away from the main app, which is also often used or developed by under aged.

**Comment by wb9688** [comment]
I think this should be revised when TeamNewPipe/NewPipe#4054 gets merged. My suggestion would be to only accept a few popular non-NSFW services in NewPipeExtractor and leave the rest to extensions.
